src/tiled/editableobject.cpp

```diff
@@ -71,6 +73,63 @@ void EditableObject::removeProperty(const QString &name)
         mObject->removeProperty(name);
 }
 
+QVariantMap EditableObject::propertiesIncludingInherited() const
+{
+    auto mergeVariantMaps = [](QVariantMap& dstMap, const QVariantMap& srcMap) {
+        for (auto iter = srcMap.constBegin(); iter != srcMap.constEnd(); ++iter)
+        {
+            dstMap.insert(iter.key(), iter.value());
+        }
+    };
+
+    QVariantMap allProperties;
+    // Insert properties into allProperties in the reverse order that
+    // Object::inheritedProperty searches them, to make sure that the
```

Comment:
I think the implementation of this function should really be moved to `Object`, so that it sits alongside `inheritedProperty`.